securedrop/tests/functional/app_navigators.py

```diff
@@ -270,3 +279,120 @@ def source_sees_flash_message(self) -> str:
+        self.nav_helper = _NavigationHelper(web_driver)
+        self.driver = web_driver
+        self.accept_languages = accept_languages
+
+    def _is_on_journalist_homepage(self):
+        return self.nav_helper.wait_for(
+            lambda: self.driver.find_element_by_css_selector("div.journalist-view-all")
+        )
+
+    def journalist_logs_in(self, username: str, password: str, otp_secret: str) -> None:
+        otp = pyotp.TOTP(otp_secret)
+        token = str(otp.now())
+        for i in range(3):
+            # Submit the login form
+            self.driver.get(f"{self._journalist_app_base_url}/login")
+            self.nav_helper.safe_send_keys_by_css_selector('input[name="username"]', username)
+            self.nav_helper.safe_send_keys_by_css_selector('input[name="password"]', password)
+            self.nav_helper.safe_send_keys_by_css_selector('input[name="token"]', token)
+            self.nav_helper.safe_click_by_css_selector('button[type="submit"]')
+
+            # Successful login should redirect to the index
+            self.nav_helper.wait_for(lambda: self.driver.find_element_by_id("link-logout"))
+            if self.driver.current_url != self._journalist_app_base_url:
+                new_token = str(otp.now())
+                while token == new_token:
+                    time.sleep(1)
+                    new_token = str(otp.now())
+                token = new_token
+            else:
+                break
+
+        assert self._is_on_journalist_homepage()
+
+    def journalist_checks_messages(self):
+        self.driver.get(self._journalist_app_base_url)
+
+        # There should be 1 collection in the list of collections
+        code_names = self.driver.find_elements_by_class_name("code-name")
+        assert 0 != len(code_names), code_names
+        assert 1 <= len(code_names), code_names
+
+        if not self.accept_languages:
+            # There should be a "1 unread" span in the sole collection entry
+            unread_span = self.driver.find_element_by_css_selector("tr.unread")
+            assert "1 unread" in unread_span.text
+
+    @staticmethod
+    def _download_content_at_url(url: str, cookies: Dict[str, str]) -> bytes:
+        r = requests.get(url, cookies=cookies, proxies=proxies_for_url(url), stream=True)
+        if r.status_code != 200:
+            raise Exception("Failed to download the data.")
+        data = b""
+        for chunk in r.iter_content(1024):
+            data += chunk
+        return data
+
+    @staticmethod
+    def _unzip_content(raw_content: bytes) -> str:
+        with tempfile.TemporaryFile() as fp:
+            fp.write(raw_content)
+            fp.seek(0)
+
+            gzf = gzip.GzipFile(mode="rb", fileobj=fp)
+            content = gzf.read()
+
+        return content.decode("utf-8")
+
+    def journalist_downloads_first_message(
+        self, encryption_mgr_to_use_for_decryption: EncryptionManager
```

Comment:
The `EncryptionManager` needs to be explicitly passed, as the one used by the `sd_servers` fixtures (ie. in a separate process) might have a different configuration than the "local" / in-process one (which, in reality, has nothing to do with the functional tests).
This is something that was confusing in the functional test suite.